recipes-extended/xen/blktap3/fix-run-time-errors-and-memory-leaks.patch

```diff
@@ -0,0 +1,270 @@
+################################################################################
+SHORT DESCRIPTION: 
+################################################################################
+Resolve run-time errors and memory leaks.
+
+################################################################################
+LONG DESCRIPTION: 
+################################################################################
+Resolve run-time errors and memory leaks.
+
+################################################################################
+CHANGELOG 
+################################################################################
+Author(s):
+Mahantesh Salimath <salimathm@ainfosec.com>
+
+################################################################################
+REMOVAL 
+################################################################################
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+Yes, should be upstreamed.
+
+################################################################################
+INTERNAL DEPENDENCIES 
+################################################################################
+
+################################################################################
+PATCHES 
+################################################################################
+Index: blktap/control/tap-ctl-free.c
+===================================================================
+--- blktap.orig/control/tap-ctl-free.c
++++ blktap/control/tap-ctl-free.c
+@@ -33,6 +33,7 @@
+ #endif
+ 
+ #include <stdio.h>
++#include <string.h>
+ #include <errno.h>
+ #include <fcntl.h>
+ #include <stdlib.h>
+@@ -48,6 +49,20 @@ tap_ctl_free(const int minor)
+ {
+ 	int fd, err;
+ 
++	char *tapdev_name, *ring_name;
++	err = asprintf(&tapdev_name, "%s%d",
++		       BLKTAP2_IO_DEVICE, minor);
++	if(err == -1){
++		err = ENOMEM;
++		return err;
++	}
++	err = asprintf(&ring_name, "%s%d",
++		       BLKTAP2_RING_DEVICE, minor);
++	if(err == -1){
++		err = ENOMEM;
++		return err;
++	}
++
+ 	fd = open(BLKTAP2_CONTROL_DEVICE, O_RDONLY);
+ 	if (fd == -1) {
+ 		EPRINTF("failed to open control device: %d\n", errno);
+@@ -57,5 +72,10 @@ tap_ctl_free(const int minor)
+ 	err = ioctl(fd, BLKTAP2_IOCTL_FREE_TAP, minor);
+ 	close(fd);
+ 
++	unlink(tapdev_name);
++	unlink(ring_name);
```

Comment:
Are these `unlink()` calls necessary?  Shouldn't udev clean the device nodes after BLKTAP2_IOCTL_FREE_TAP above?

Reply:
I think so, last time I checked those were not removed. Will check again and update accordingly.

Reply:
I checked again and the nodes are not removed by default. BLKTAP2_IOCTL_FREE_TAP is used by kernel module only to free its resources.

Reply:
Okay.  Thanks for checking.